docs/rfcs/1000-shamir-based-recovery.md

```diff
@@ -730,6 +731,121 @@ The claimer gets access to `reveal_token` and `data_key`, it can then retrieve `
 Then `ciphered_data` can be decrypted with `data_key`. From then on, the recovery works
 just like the recovery device system (see `parsec core import_recovery_device` CLI).
 
+## 4 - Delete a shamir setup
+
+Authenticated API:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "shamir_recovery_deletion",
+            "fields": [
+                {
+                    "name": "certificate",
+                    // Contains a ShamirRecoveryDeletionCertificate
+                    "type": "bytes"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok"
+            },
+            {
+                // Cannot deserialize data into the expected certificate
+                "status": "invalid_data"
+            },
+            {
+                // Given timestamp does not match previous timestamps
+                "status": "incoherent_previous_timestamp"
+            }
+            {
+                // Nothing to delete
+                "status": "no_setup_to_delete"
+            },
+            {
+                // Returned if the timestamp in the certificate is too far away compared
+                // to server clock.
+                "status": "timestamp_out_of_ballpark",
+                "fields": [
+                    {
+                        "name": "ballpark_client_early_offset",
+                        "type": "Float"
+                    },
+                    {
+                        "name": "ballpark_client_late_offset",
+                        "type": "Float"
+                    },
+                    {
+                        "name": "server_timestamp",
+                        "type": "DateTime"
+                    },
+                    {
+                        "name": "client_timestamp",
+                        "type": "DateTime"
+                    }
+                ]
+            },
+            {
+                // Returned if another certificate or vlob in the server has a timestamp
+                // posterior or equal to our current one.
+                "status": "require_greater_timestamp",
+                "fields": [
+                    {
+                        "name": "strictly_greater_than",
+                        "type": "DateTime"
+                    }
+                ]
+            }
+        ],
+    }
+]
+```
+The deletion certificate
+
+```json5
+{
+    "label": "ShamirRecoveryDeletionCertificate",
+    "type": "shamir_recovery_deletion_certificate",
+    "other_fields": [
+        {
+            "name": "author",
+            "type": "DeviceID"
+        },
+        {
+            "name": "timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "previous_timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "share_recipients",
+            "type": "List<UserId>"
```

Comment:
There is no "share recipients mismatch" status in `shamir_recovery_deletion`, is that right?

Reply:
Yes indeed, we can remove that.

Reply:
After thinking about it a bit more, I'd rather keep the share recipients in the certificate, even if we could retrieve the information, because it's a certificate we went them to get.

Reply:
Sorry for not being clear, I meant that we probably need to add a "share recipients mismatch" status to `shamir_recovery_deletion` to return when this field doesn't match what is in the previous certificate.